RIOT-OS/RIOT
# cpu/esp32: allow external 32 kHz crystal for the RTC hardware timer

### Contribution description

This PR changes the RTC low-level driver to allow the use of the RTC hardware timer with an external 32.768 kHz crystal.

The RTC hardware timer of ESP32 can be clocked with either an external 32.768 kHz crystal or an internal adjustable 150 kHz RC oscillator. If the the external 32.768 kHz crystal is not available (as on most boards), the internal 150 kHz RC oscillator is used automatically. However, since this internal 150 kHz RC oscillator is not very accurate, the RTC low-level driver uses by default the PLL-controlled 64-bit microsecond system timer to emulate the RTC timer.

To allow the use of the RTC hardware timer for boards with an external 32 kHz crystal, the pseudomodules `esp_rtc_timer` and `esp_rtc_timer_32k` have been defined to control which timer is used for RTC as following:

- default
    Use the emulated RTC timer. The RTC hardware timer with the internal RC 150 kHz oscillator is only used in deep sleep mode and during a reset.

- `esp_rtc_timer`
     Use always the RTC hardware timer with the internal 150 kHz RC oscillator.

- `esp_rtc_timer_32k`
     Use the RTC hardware timer with the external 32.768 kHz crystal. If the external 32.768 kHz crystal is not  available, the RTC hardware timer is used with the internal 150 kHz RC oscillator.

Furthermore, module `esp_rtc_timer_32k` is enabled by default for board `esp32-wrover-kit` by this PR.

This PR requires the bootloader images and the automatic XTAL clock detection from PR #13059.

### Testing procedure

1. Use a board with and a board without external 32.768 kHz crystal. Flash and test both boards with
    ```
    LOG_LEVEL=4 make BOARD=esp32-wroom-32 -C test/periph_rtc flash test
    ```
    and
    ```
    LOG_LEVEL=4 USEMODULE=esp_rtc_timer_32 make BOARD=esp32-wroom-32 -C test/periph_rtc flash test
    ```
   All tests should work as expected. Module `esp_rtc_timer_32` depends on module `esp_rtc_timer` so that all cases are covered.

2. Use a terminal program, reset the boards and check for following output during startup. `SLOW` is the clock used for RTC timer:

    - with enabled `esp_rtc_timer_32k` and the board with 32.768 kHz crystal:
       ```
       Used clocks in Hz: CPU=80000000 APB=80000000 XTAL=40000000 FAST=8000000 SLOW=32768
       ```
    - with enabled `esp_rtc_timer_32k` and the board w/o 32.768 kHz crystal:
       ```
       RTC: Not found External 32 kHz XTAL. Switching to Internal 150 kHz RC chain
       Used clocks in Hz: CPU=80000000 APB=80000000 XTAL=40000000 FAST=8000000 SLOW=150000
       ```
    - w/o module `esp_rtc_timer_32k` and any board:
       ```
       Used clocks in Hz: CPU=80000000 APB=80000000 XTAL=40000000 FAST=8000000 SLOW=150000
       ```

### Issues/PRs references

Depends on PR #13059
